packages/TorneloScoresheet/src/components/TimePickerSheet/TimePickerSheet.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import { View } from 'react-native';
+import DatePicker from 'react-native-date-picker';
+import { GameTime } from '../../types/ChessMove';
+import PrimaryButton from '../PrimaryButton/PrimaryButton';
+import Sheet from '../Sheet/Sheet';
+import { styles } from './style';
+
+export type TimePickerSheetProps = {
+  gameTime: GameTime;
+  visible: boolean;
+  dismiss: () => void;
+  setGameTime: (gameTime: GameTime | undefined) => void;
+};
+
+const TimePickerSheet: React.FC<TimePickerSheetProps> = ({
+  gameTime,
+  visible,
+  dismiss,
+  setGameTime,
+}) => {
+  // set day/month/year not important
+  // date should be + 1 on hours for it to display properly,
+  // this is taken into account when setting the date
+  const [date, setDate] = useState(
+    new Date(1, 1, 1, gameTime.hours + 1, gameTime.minutes, 0),
```

Comment:
I think it makes sense to make the default time the current time? Otherwise they may have to scroll through like 30 minutes everytime

Reply:
Yes, that's what this code does, the `TimePickerSheet` component takes in the gameTime as an argument.
This gameTime is either:
- The current time saved on the Ply or
- The current time since the start of the game if there is no saved time on that Ply

The game time is being measured in hours and minutes.

Reply:
Right, well I was just testing this and it came up with "1:00" even though the time was "1:38" or something, so there may be an issue somewhere? Does it work for you?

Reply:
Hang on, I see what you mean, that means this move is one hour since the start of the game. Shouldn't it have been "0:01" or something then?

Reply:
I'm not sure, but I think it would be clearer to have that UI be entering the time that the move happened? Might be one to check with David/Simon

Reply:
Yes the time is the time since the start of the game, i checked the rules.
So you log the time since the game started.
You should see 1 minute, did you see 1 hour?
Was the emulator left open for a long time?

Reply:
Okay cool! Nice work.

Yep I saw one hour, the emulator had just been opened. In the code snippet this original comment was on it says:

```
new Date(1, 1, 1, gameTime.hours + 1, gameTime.minutes, 0)
```

Maybe it's the `+ 1` on the gameTime hours?

Reply:
Yeh so this must be an Ios, android compatibility thing,
On android, it wont display the time correctly if i dont have that +1 it shows me 24...
Since we're developing for ios, i guess ill change it to work there?

Reply:
Yeah nah, it crashes now. I'll try and diagnose